File: LearningJava/Polymorphism/BasePlusCommissionEmployee.java
public class BasePlusCommissionEmployee extends CommissionEmployee
{
  private double baseSalary;
  
  public BasePlusCommissionEmployee(String firstName, String lastName, String socialSecurityNumber, 
    double grossSales, double commissionRate, double baseSalary)
  {
    super(firstName, lastName, socialSecurityNumber, grossSales, commissionRate);
    
    this.baseSalary = baseSalary;
  }
  
  @Override
  public double earnings()
  {
    return 0;
  }
  
  @Override
  public String toString()
  {
    return String.format("%sbaseSalary = %.2f%n", super.toString(), baseSalary);
  }
 
  
}

File: C++Programming/dynamic_cast/polymorphism/BasePlusCommissionEmployee.cpp
//
//  BasePlusCommissionEmployee.cpp
//  polymorphism
//
//  Created by Vijay Kumar on 9/18/13.
//  Copyright (c) 2013 Vijay Kumar. All rights reserved.
//

#include "Employee.h"
#include "CommissionEmployee.h"
#include "BasePlusCommissionEmployee.h"

BasePlusCommissionEmployee::BasePlusCommissionEmployee(
    const string& first_name, const string& last_name,
    const string& ssn, const double gross_sales,
    const double commission_rate, const double base_salary)
:CommissionEmployee(first_name, last_name, ssn, gross_sales, commission_rate)
{
  baseSalary = base_salary;
  
}

double BasePlusCommissionEmployee::earnings() const
{
  return baseSalary + CommissionEmployee::earnings();
}

void BasePlusCommissionEmployee::print() const
{
  CommissionEmployee::print();
  cout << "Base Salary = " << baseSalary << endl;
}

void BasePlusCommissionEmployee::setSalary(double salary_passed)
{
  baseSalary = salary_passed;
}

double BasePlusCommissionEmployee::getSalary() const
{
  return baseSalary;
}